### DISCUSSION TITLE
Long string added to figure cross reference

### DISCUSSION THREAD

**Discussion by n8craig** [quarto-dev/quarto-cli#872] [Q&A] [answered]
I have a document with figures and cross references. Unfortunately, it is long and I can’t seem to isolate the issue in a reproducible way. However, there’s something curious with the cross references. Assuming the following chunk and cross reference:

```r
#| label: fig-foo
```

`@fig-foo`

In the rendered document the cross reference does not work. Looking at the preview page I saw the link to the figure was:

http://localhost:3749/path/to/file/#fig-fooD08295A6-16DC-499D-85A8-8BA656E013A2

If I cross reference to `@fig-fooD08295A6-16DC-499D-85A8-8BA656E013A2` it works and shows Figure 1.

The string `D08295A6-16DC-499D-85A8-8BA656E013A2` isn’t in the document. 

I’ve changed the file from .qmd to .Rmd, I’ve renamed it, and put it in different locations. That long string is still sneaking in somehow.

Anyone have a sense of what is going on and how I can rid the `D08295A6-16DC-499D-85A8-8BA656E013A2`  from the cross reference?

**Comment by cscheid** [comment]
I think something in our end is getting confused. That string you're seeing is a UUID we use internally that should be getting cleared out and isn't. In the meantime, can you try the following syntax instead?

    ```{r}
    #| label: fig-foo
    ```

**Reply by n8craig** [reply]
Facepalm, I initially typed it incorrectly here. It is correct in the document and I've fixed the typo in my question. Sorry for that.

Thank you for clarifying the UUID. For the short term, is that number stable? In other words, at this point for the short term I'm thinking of just adding it to all of the document's cross reffs but wasn't sure if it could change unexpectedly. And for some reason this issue is just this one document which is perplexing.

**Reply by cscheid** [reply]
That line of code hasn't changed in two years: https://github.com/quarto-dev/quarto-cli/blame/946b31a703bda35218c26323d1c2349f0cca08a5/src/resources/rmd/hooks.R#L874

**Reply by cscheid** [reply]
With that said, this is 100% our bug so if you can find a small reproducible example, please share with us!

**Reply by jjallaire** [reply]
Thanks, that repros for me! I'll fix this ASAP.

**Comment by n8craig** [comment]
I pasted this into a new document and got the same result. It should be reproducible.

---

```{r load-libraries}

library(readr)
library(tidyverse)
library(lubridate)
library(ggplot2)
library(plotly)
library(janitor)
library(knitr)

options(digits=3)
opts_chunk$set(results = 'asis',
               comment = NA,
               prompt = FALSE,
               cache = FALSE)

# Turn off scientific notation
options(digits=5, scipen=15)
```

```{r load-dataset}

# Load Dataset
df_summary <- read_csv("https://raw.githubusercontent.com/n8craig/ICE-COVID-19/main/data/covid_summaries.csv", 
    col_types = cols(Date = col_date(format = "%Y-%m-%d"), 
        `Total Deaths` = col_double(), `Total COVID-19 Confirmed in Custody` = col_double())) %>% 
  clean_names()
```

@fig-ice-detention-current

```{r}
#| label: fig-ice-detention-current
#| fig-cap: "ICE detention population plotted over time. Data source: https://www.ice.gov/coronavirus"
#| column: body-outset

# ICE Detention Population Over Time
p <- ggplot(df_summary, aes(x=date, y=total_detained))+
  geom_line()+
  geom_point()+
  geom_smooth()+
  labs(
    title = paste("ICE Detention Population as of", Sys.Date()),
    x = "Date",
    y = "Total Detained"
  )

ggplotly(p)
```

@fig-total-confirmed-covid-19

```{r}
#| label: fig-total-confirmed-covid-19
#| fig-cap: "Total number of active COVID-19 cases plotted over time. Data source: https://www.ice.gov/coronavirus"
#| column: body-outset

# Total Confirmed COVID-19 Over Time
p <- ggplot(df_summary, aes(x=date, y=total_covid_19_confirmed_in_custody))+
  geom_line()+
  geom_smooth()+
   labs(
    title = "Total Number of Confirmed COVID-19 Cases in ICE Detention",
    x = "Date",
    y = "Total COVID-19 Confirmed in Custody"
  )

ggplotly(p)
```

**Comment by jjallaire** [comment, marked as answer]
Okay, I've resolved the issue here: https://github.com/quarto-dev/quarto-cli/commit/d4e12dc0b952f9e1534248cdddd03bc25e9c2d94

Your example should now behave as you expect (although pls. note that crossrefs are case-sensitive so you need to change `@fig-Total-Confirmed-COVID-19` to `@fig-total-confirmed-covid-19`)

**Reply by n8craig** [reply]
Thank you very much. It works perfectly now both on the example and the larger document. I edited my example above to remove the typo.
